### PR TITLE
Don't panic with large int header labels

### PR DESCRIPTION
**Pull request by qmuntal**
Found via fuzzing.

`ProtectedHeader.UnmarshalCBOR` panics if a map label value does not fit in a `int64` because `cbor` converts it to a `big.Int`, which can't be used as map key because it is not comparable.

We don't support those large labels anyway, so we better error out than panic.

@shizhMSFT

### CODE REVIEW

File: headers_test.go [hunk 1]
```diff
@@ -251,6 +251,13 @@ func TestProtectedHeader_UnmarshalCBOR(t *testing.T) {
 			},
 			wantErr: true,
 		},
+		{
+			name: "int map key too large",
+			data: []byte{
+				0x4b, 0xa1, 0x3b, 0x83, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30,
```
Comment (shizhMSFT) [inline headers_test.go:257]: It is better to have a full header data.
```suggestion
				0x4c, 0xa1, 0x3b, 0x83, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x00
```

File: headers_test.go [hunk 2]
```diff
@@ -251,6 +251,13 @@ func TestProtectedHeader_UnmarshalCBOR(t *testing.T) {
 			},
 			wantErr: true,
 		},
+		{
+			name: "int map key too large",
```
Comment (shizhMSFT) [inline headers_test.go:255]: We also need this test for unprotected header.

**Comment by jaredpar** [comment]
> Found via fuzzing.

What tool were you fuzzing with here?

File: headers_test.go (hunk 2 above)
Comment (qmuntal) [reply headers_test.go:255]: Done!

File: headers_test.go [hunk 3]
```diff
@@ -596,6 +603,13 @@ func TestUnprotectedHeader_UnmarshalCBOR(t *testing.T) {
...
+			data: []byte{
+				0xa1, 0x3b, 0x83, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30,
```
Comment (shizhMSFT) [inline headers_test.go:609]: Same here. We want a complete _label -> value_ pair in the map.
```suggestion
				0xa1, 0x3b, 0x83, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x30, 0x00,
```

Comment (qmuntal) [reply headers_test.go:609]: Hmm, when putting the original sequence in [cbor.me](https://cbor.me/?bytes=21((13.68.16.48.48.48.16.48.88.30.48.48.48.48.48.48.48.48.48.48.48.48.48.48.48.48.48.48.48.48.48.48.48.48.48.48.48.48.48.48.84.48.48.48.48.48.48.48.48.48.48.48.48.48.48.48.48.48.48.48.48.65.48) it tells me that the byte chunk represents a complete map, and that the tailing `0x00` is unused.

Comment (shizhMSFT) [reply headers_test.go:609]: Ah. My mistake. I counted it wrongly. Could you also revert the change at line 257?

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**